### ISSUE TITLE
Implement stable Backpropagation and Simple Optimisation routine

### ISSUE THREAD

**Issue by jjerphan** [jjerphan/joml#1]
**Goals:**
 - Work out the back-propagation algorithm
 - Implement simple gradient descent with an arbitrary constant step size.
